tests/TurboMqtt.Tests/End2End/TcpMqtt311End2EndSpecs.cs

```diff
@@ -111,14 +111,18 @@ public async Task ShouldTerminateClientAfterMultipleFailedConnectionAttempts()
     [Fact]
     public async Task ShouldFailToConnectToNonExistentServer()
     {
-        var updatedTcpOptions = new MqttClientTcpOptions("localhost", 21884);
+        var updatedTcpOptions = new MqttClientTcpOptions("localhost", 21884)
+        {
+            MaxReconnectAttempts = 0
```

Comment:
If an initial connection fails, the `IMqttClient` is dead anyway since we could never establish a connection to the host. It's the caller's responsibility to recreate it.